SparkPost/2web2ui
# FE-1218 - Pagination for domain list pages

### What Changed
 - Client side paging for the domains/sending, domains/bounce, and domains/tracking pages
 - react-table useTable instead of our own custom useTable
 - Default sort for trackingDomains wasn't working A - Z - fixed with the use of react-table useTable
 - Rows with dates that were undefined | null are now sorting to the end Newest - Oldest
 - Numbers now fall to the end of A - Z
 - Sorting, Filtering, and Paging interactions all work together
 - Landing the page with url params still applies them to the UI
 - Updating the UI filters still updates the URL

### How To Test
http://localhost:3100/domains/list/sending
http://localhost:3100/domains/list/tracking

1. Sorting, Filtering, Paging all work together
2. URL is synced with applied filters 
3. Filters apply on page load based on URL params

1a. Go to page 2 and filter -> paging should be reset to first page
2a. Default sort on page load should be correct (dates that are undefined/null fall to end of list & numbers fall to the end of the list for [a - z]) 
3. Domain status filter is applied as an && filter. 

### To Do
- [ ] Review / Get Team Feedback


### Thoughts for the future
- Decide on a deprecation strategy

- Debounce the domain name input so it's a little smoother (sometimes it lags a little but I admit that might be because of the multiple things happening in the event handler)
